src/Microsoft.Cci.Extensions/Writers/CSharp/CSDeclarationWriter.Methods.cs

```diff
@@ -206,7 +206,10 @@ private void WriteMethodBody(IMethodDefinition method)
                 Write("throw new ");
                 if (_forCompilationIncludeGlobalprefix)
                     Write("global::");
-                Write("System.PlatformNotSupportedException(); ");
+                if(_platformNotSupportedExceptionMessage == null)
+                    Write("System.PlatformNotSupportedException();");
+                else
+                    Write($"System.PlatformNotSupportedException(\"{_platformNotSupportedExceptionMessage}\"); ");
```

Comment:
Sure if you can get quotes to stick through the commandline parameter parsing you could do that and just avoid adding quotes in GenAPI